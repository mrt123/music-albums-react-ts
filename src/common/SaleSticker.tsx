const stickerStyle = {
  color: "black",
  background: "yellow",
  fontWeight: "800",
  fontFamily: "'Courier New', monospace",
  border: "2px solid black",
  padding: "3px",
};

interface SaleStickerProps {
  show: boolean;
  value?: number;
}

const SaleSticker: React.FC<SaleStickerProps> = ({ show, value }) => {
  if (!show) return null;
  return <div style={stickerStyle}>{value}% Off!</div>;
};

export default SaleSticker;
