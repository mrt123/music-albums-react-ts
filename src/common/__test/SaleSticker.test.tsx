import { render } from "@testing-library/react";
import SaleSticker from "../SaleSticker";

describe("SaleSticker.tsx", () => {
  it("matches snapshot when show is true", () => {
    const { container } = render(
      <SaleSticker show={true} value={40}></SaleSticker>
    );
    expect(container).toMatchSnapshot();
  });

  describe("renders sale sticker when appropraite", () => {
    it("shows sale sticker with 40% off", () => {
      const { queryByText } = render(
        <SaleSticker show={true} value={40}></SaleSticker>
      );

      const textQueryResult = queryByText("40% Off!");
      expect(textQueryResult).toBeInTheDocument();
    });

    it("does NOT shows sale sticker with 40% off", () => {
      const { queryByText } = render(
        <SaleSticker show={false} value={40}></SaleSticker>
      );

      const textQueryResult = queryByText("40% Off!");
      expect(textQueryResult).toEqual(null);
    });
  });
});
