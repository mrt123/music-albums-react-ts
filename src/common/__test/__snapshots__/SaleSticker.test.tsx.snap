// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SaleSticker.tsx matches snapshot when show is true 1`] = `
<div>
  <div
    style="color: black; background: yellow; font-weight: 800; font-family: 'Courier New', monospace; border: 2px solid black; padding: 3px;"
  >
    40
    % Off!
  </div>
</div>
`;
